commands/ApiController.php

```diff
@@ -113,8 +115,61 @@ public function actionGenerate($version)
         return 0;
     }
 
+    public function actionExtension($extensionName)
+    {
+        $extension = Extension::find()->where(['name' => $extensionName])->active()->one();
+
+        if ($extension === null) {
+            $this->stderr("Unknown extension $extensionName.\n", Console::FG_RED);
+            return 1;
+        }
+        $this->_extension = $extension;
+
+        $targetPath = Yii::getAlias("@app/data/extensions/$extensionName");
+
+        // TODO determine versions
+        $versions = ['2.0'];
+
+        foreach($versions as $version) {
+
+            $this->version = $version;
+
+            $sourcePath = Yii::getAlias("@app/data/extensions/$extensionName/$version");
+
+            if (is_dir("$sourcePath/src")) {
+                $source = ["$sourcePath/src"];
+            } else {
+                $source = [$sourcePath];
+            }
+            $target = "$targetPath/api-$version";
+            $this->guide = "/extension/$extensionName/doc/guide/{$this->version}/en";
+
+            $this->stdout("Start generating $extensionName API $version...\n");
+            $this->template = 'extension';
+            $this->actionIndex($source, $target);
+
+            $this->stdout("Start generating $extensionName API $version JSON Info...\n");
+            $this->template = 'json';
+            $this->actionIndex($source, $target);
+            $this->splitJson($target);
+
+            $this->stdout("Finished $extensionName API $version.\n\n", Console::FG_GREEN);
+        }
+        file_put_contents("$targetPath/api.json", Json::encode($versions));
+
+        return 0;
```

Comment:
https://github.com/yiisoft/yii2/blob/master/framework/console/ExitCode.php?

Reply:
this code base is older than that class :) but I'm happy to add it.